tests/unit/amo/components/TestAddon.js

```diff
@@ -255,6 +255,24 @@ describe('Addon', () => {
     expect(root.find(NotFound)).toHaveLength(1);
   });
 
+  it('renders NotFound page for unauthorized add-on - 401 error', () => {
+    const id = 'error-handler-id';
+    const store = dispatchClientMetadata().store;
+
+    const error = createApiError({
+      response: { status: 401 },
+      apiURL: 'https://some/api/endpoint',
+      jsonResponse: { message: 'Authentication credentials were not provided.' },
+    });
+    store.dispatch(setError({ id, error }));
+    const capturedError = store.getState().errors[id];
```

Comment:
I know you moved it per request but I think you should put this line back, including the comment:

```js
// This makes sure the error was dispatched to state correctly.
expect(capturedError).toBeTruthy();
```

This pattern is a bit clunky (sorry) but I think it's the right way to do it for now.